packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  HttpRequest,
+  HttpResponse,
+  ApiLoggerInterface,
+  LoggerInterface,
+  Level,
+} from '../coreInterfaces';
+import { LoggingOptions } from './loggerOptions';
+import { NullLogger } from './nullLogger';
+
+export class ApiLogger implements ApiLoggerInterface {
+  private readonly _logger: LoggerInterface;
+
+  constructor(loggingOp: LoggingOptions) {
+    this._logger = loggingOp.logger ?? new NullLogger();
```

Comment:
ApiLogger should not take a direct dependency on the NullLogger. The logger instance should be injected. If the concern is that the constructor parameters allow null/undefined to be passed, change the parameters types to disallow that from happening. Consider adding null/undefined checks to the constructor as well.